src/HDFFile.cpp

```diff
@@ -9,889 +9,66 @@
-    std::copy_n(String.data(), String.size(), std::back_inserter(Buffer));
-  }
-};
-
-template <typename DataHandler>
-static std::vector<typename DataHandler::DataType>
-populateBlob(nlohmann::json const &ValueJson, size_t const GoalSize,
-             size_t const ItemLength = 0) {
-  using DataType = typename DataHandler::DataType;
-  std::vector<DataType> Buffer;
-  if (ValueJson.is_array()) {
-    std::stack<StackItem> Stack;
-    Stack.emplace(ValueJson);
-    while (!Stack.empty()) {
-      if (Stack.size() > MAX_DIMENSIONS_OF_ARRAY) {
-        break;
-      }
-      if (Stack.top().exhausted()) {
-        Stack.pop();
-        continue;
-      }
-      auto const &Value = Stack.top().value();
-      if (Value.is_array()) {
-        Stack.top().inc();
-        Stack.emplace(Value);
-      } else {
-        Stack.top().inc();
-        DataHandler::append(Buffer, Value, ItemLength);
-      }
-    }
-  } else {
-    DataHandler::append(Buffer, ValueJson, ItemLength);
-  }
-  if (GoalSize != 0 && Buffer.size() != GoalSize) {
-    auto What =
-        fmt::format("Failed to populate numeric blob, size mismatch: {} != {}",
-                    Buffer.size(), GoalSize);
-    std::throw_with_nested(std::runtime_error(What));
-  }
-  return Buffer;
-}
-
-template <typename T>
-static void writeAttrNumeric(hdf5::node::Node const &Node,
-                             std::string const &Name,
-                             nlohmann::json const &Value) {
-  size_t Length = 1;
-  if (Value.is_array()) {
-    Length = Value.size();
-  }
-  try {
-    auto ValueData = populateBlob<NumericItemHandler<T>>(Value, Length);
-    try {
-      if (Value.is_array()) {
-        writeAttribute(Node, Name, ValueData);
-      } else {
-        writeAttribute(Node, Name, ValueData[0]);
-      }
-    } catch (std::exception const &E) {
-      std::throw_with_nested(std::runtime_error(
-          fmt::format("Failed write for numeric attribute {} in {}: {}", Name,
-                      std::string(Node.link().path()), E.what())));
-    }
-  } catch (std::exception const &E) {
-    std::throw_with_nested(std::runtime_error(
-        fmt::format("Can not populate blob for attribute {} in {}: {}", Name,
-                    std::string(Node.link().path()), E.what())));
-  }
-}
-
-HDFFile::HDFFile() {
+using HDFOperations::createHDFStructures;
+using HDFOperations::writeAttributesIfPresent;
+using HDFOperations::writeHDFISO8601AttributeCurrentTime;
+using HDFOperations::writeStringAttribute;
+
+HDFFile::HDFFile(std::string const &FileName,
+                 nlohmann::json const &NexusStructure,
+                 std::vector<StreamHDFInfo> &StreamHDFInfo)
+    : H5FileName(FileName) {
 // Keep this.  Will be used later to test against different lib versions
 #if H5_VERSION_GE(1, 8, 0) && H5_VERSION_LE(1, 10, 99)
   unsigned int maj, min, rel;
   H5get_libversion(&maj, &min, &rel);
 #else
   static_assert(false, "Unexpected HDF version");
 #endif
```

Comment:
I suggest we take the opportunity to get rid of this unused code.
```suggestion
```

Reply:
👍